esp/services/ws_dfu/ws_dfuXRefService.cpp

```diff
@@ -724,24 +720,58 @@ bool CWsDfuXRefEx::onDFUXRefUnusedFiles(IEspContext &context, IEspDFUXRefUnusedF
     Owned<IPropertyTree> controlXrefInfo = sendRoxieControlQuery(sock, "<control:getQueryXrefInfo/>", ROXIECONTROLXREFTIMEOUT);
     if (!controlXrefInfo)
         throw MakeStringExceptionDirect(ECLWATCH_INTERNAL_ERROR, "roxie cluster, not responding.");
-    MapStringTo<bool> usedFileMap;
     Owned<IPropertyTreeIterator> roxieFiles = controlXrefInfo->getElements("//File");
     ForEach(*roxieFiles)
         addLfnToUsedFileMap(usedFileMap, roxieFiles->query().queryProp("@name"));
 
-    if (req.getCheckPackageMaps())
+    if (checkPackageMaps)
         addUsedFilesFromPackageMaps(usedFileMap, process);
+}
+
+bool CWsDfuXRefEx::onDFUXRefUnusedFiles(IEspContext &context, IEspDFUXRefUnusedFilesRequest &req, IEspDFUXRefUnusedFilesResponse &resp)
+{
+    const char *process = req.getProcessCluster();
+    StringArray &processList = req.getProcessClusterList();
+    if (isEmptyString(process) && !processList.length())
+        throw MakeStringExceptionDirect(ECLWATCH_INVALID_INPUT, "process cluster not specified.");
+    bool checkPackageMaps = req.getCheckPackageMaps();
+    MapStringTo<bool> usedFileMap;
+    if (!isEmptyString(process))
+        getRoxieFiles(process, checkPackageMaps, usedFileMap);
+    ForEachItemIn(i, processList)
+    {
+        getRoxieFiles(processList.item(i), checkPackageMaps, usedFileMap);
+    }
+
+    StringArray &checkPlanes = req.getCheckPlanes();
     if (!req.getGetFileDetails())
     {
         StringArray unusedFiles;
-        findUnusedFilesInDFS(unusedFiles, process, usedFileMap);
+        if (checkPlanes.length())
+        {
+            ForEachItemIn(idx, checkPlanes)
```

Comment:
Efficiency: If it becomes common to pass multiple planes we may want to revisit this so that it passes a list.  Some of the underlying functions support a comma separated list, but I'm not sure if they all do.